### PR TITLE
makes make everyone random not the worst button of all time

### LINKED ISSUE #20168: Make Everyone Random does it in the shittiest way possible
**Issue Description**:

Make everyone random button, the one that makes people have random characters when round start, does THIS.
```
		client.prefs.active_character.randomise()
		client.prefs.active_character.real_name = random_name(client.prefs.active_character.gender)
```

**What did you expect to happen**: 

not mess with the PLAYERS ACTIVE CHARACTER (note it does not save as far as I can tell but it DOES give people / me a heart attack

**What happened instead**:

it gave me a heart attack

**Why is this bad/What are the consequences**:

I like not worrying about deleting peoples characters.

**Steps to reproduce the problem**:

press make everyone random.
Have the round start.
Suffer.

**When did the problem start happening**:

probably forever.

**Extra information**:

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Make everyone random now works without randomizing the players *currently saved character in preferences that will fuck them if they save it*

## Why It's Good For The Game

the button being usable is good

fix #20168

## Testing

confirmed it scrambled, and that the save was not changed

## Changelog
:cl:
fix: make everyone random will no longer fuck up peoples saved characters so it can be used.
/:cl:

### CODE REVIEW

**Comment by Qwertytoforty** [comment]
I am tempted to make it random species, though would make it a tweak

**Review by Sirryan2002 — APPROVED** [review]

File: code/modules/mob/new_player/new_player.dm [hunk 1]
```diff
@@ -544,12 +544,17 @@
...
-		client.prefs.active_character.real_name = random_name(client.prefs.active_character.gender)
 	client.prefs.active_character.copy_to(new_character)
-
+	if(SSticker.random_players)
+		var/mob/living/carbon/human/H = new_character
+		scramble(1, H, 100)
+		H.real_name = random_name(H.gender, H.dna.species.name) //Give them a name that makes sense for their species.
+		H.sync_organ_dna(assimilate = 1)
+		H.update_body()
+		H.reset_hair() //No more winding up with hairstyles you're not supposed to have, and blowing your cover.
```
Comment (lewcc) [inline code/modules/mob/new_player/new_player.dm:554]: I take it some of these comments are from the DNA scrambler? Could do without them

Comment (Qwertytoforty) [reply code/modules/mob/new_player/new_player.dm:554]: can do

**Review by AffectedArc07 — APPROVED** [review]
Thank fuck for that

**Review by lewcc — APPROVED** [review]

**Pull request merged**